### PR TITLE
Data population architecture for secondary view controller

### PR DESCRIPTION
**Pull request by lukeandrews239**
- This PR implements architecture for listening for candidate votes, and subsequently voting and requesting updated information with which we instantiate the secondary Poll display view controller.
- Several TODOs and comments to myself, since this is dependent on a bit of the work in the PR of @meieri 
- Will need to get that approved and merged, and I'll rebase off that

### CODE REVIEW

File: VoiceOfAmerica/ViewControllers/VoteViewController.swift [hunk 1]
```diff
@@ -76,3 +72,26 @@ class VoteViewController: UIViewController, UITableViewDelegate, UITableViewData
...
 
+extension VoteViewController {
+    // This function will create an alertView that seals one's vote
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        // Do something with AlertView, which has an action completion to call didVote
+        let finalVoteAlert = UIAlertController(title: "Cast Vote?", message: "You can only vote once, all votes are final!", preferredStyle: .alert)
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel, handler: nil)
+        let castAction = UIAlertAction(title: "Vote", style: .default) { [weak self] action in
+            // We'll need to index the candidate array here with indexPath.row get the right candidate
+            self?.didVoteForCandidate(candidate: nil)
```
Comment (meieri) [inline VoiceOfAmerica/ViewControllers/VoteViewController.swift]: Do it then baby.

File: VoiceOfAmerica/Model/CandidateModel.swift [hunk 2]
```diff
@@ -48,7 +50,11 @@ class CandidateModel {
                     weakSelf.candidates.append(Candidate(name: name, bio: bio ?? "Dic/Server mismatch", face: UIImageView(), votes: voteTally))
                 }
             }
-            weakSelf.delegate?.didFinishLoadingData()
+            if let completionToExecute = completion {
```
Comment (meieri) [inline VoiceOfAmerica/Model/CandidateModel.swift]: #pp Rename to dataCompletion

File: VoiceOfAmerica/Model/CandidateModel.swift [hunk 3]
```diff
@@ -18,11 +18,13 @@ class CandidateModel {
 
     var delegate: CoordinatorDelegate?
 
-    init() {
-        makeCandidates()
+    typealias DataCompletion = ([Candidate], CandidateDelegate) -> ()
```
Comment (meieri) [inline VoiceOfAmerica/Model/CandidateModel.swift]: cool

File: VoiceOfAmerica/ViewControllers/PollDataViewController.swift [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    // Placeholders to compile
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return 0
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        return UITableViewCell()
+    }
+
+}
```
Comment (meieri) [inline VoiceOfAmerica/ViewControllers/PollDataViewController.swift]: sweet

File: VoiceOfAmerica/ViewControllers/VoteViewController.swift (hunk 1 above)
Comment (lukeandrews239) [reply VoiceOfAmerica/ViewControllers/VoteViewController.swift]: Done

File: VoiceOfAmerica/Coordinators/CoordinatorDelegate.swift [hunk 5]
```diff
@@ -11,6 +11,6 @@ import UIKit
 
 protocol CoordinatorDelegate: AnyObject {
     // Interface to limit access of View Controllers that neccesitate callbacks to the coordinator
-    func didVote(candidate: String)
+    func didVote(candidate: Candidate)
     func didFinishLoadingData()
```
Comment (BrianSemiglia) [inline VoiceOfAmerica/Coordinators/CoordinatorDelegate.swift]: Needs caller as first argument: `coordinateDelegateDidFinishLoadingData(_ caller: CoordinatorDelgate)`

File: VoiceOfAmerica/Coordinators/FlowCoordinator.swift [hunk 6]
```diff
@@ -23,18 +24,40 @@ class FlowCoordinator: Coordinator {
...
+        let model = CandidateModel(dataCompletion: nil)
         self.model = model
         voteVC.delegate = self
         voteVC.model = model
         model.delegate = self
         self.voteViewController = voteVC
         self.navigationController.pushViewController(voteVC, animated: true)
     }
+
+    private func voteCast(candidateList: [Candidate]) {
```
Comment (meieri) [inline VoiceOfAmerica/Coordinators/FlowCoordinator.swift]: #pp rename either voteCast or didVote to differentiate between the two

**Pull request merged**